src/main/java/com/celements/web/FileAction.java

```diff
@@ -312,8 +317,11 @@ public boolean renderFileFromObjectField(String filename, XWikiDocument doc,
   public boolean renderFileFromAttachment(String filename, XWikiDocument doc,
       XWikiContext context) throws IOException, XWikiException {
     _Logger.debug("... as attachment");
-    XWikiAttachment attachment = doc.getAttachment(filename);
-    if (attachment != null) {
+    try {
+      IAttFileNameMatcherRole matcher = Utils.getComponent(IAttFileNameMatcherRole.class);
+      matcher.setFileNamePattern(filename);
+      XWikiAttachment attachment = getAttachmentService().getAttachmentFirstNameMatch(doc,
```

Comment:
this replacement looks wrong to me, because the doc.getAttachment FIRST checks for name equals and then only returns similar results. Yet, getAttachmentFirstNameMatch may return a near match eventhough there would be an exact filename match attachment.